src/providers/CartInfoProvider.js

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useState } from 'react'
+
+export const CartInfoContext = createContext()
+
+
+// all info relating to the cart
+export const CartInfoProvider = (props) => {
+
+    const confirmedCartOrders = JSON.parse(localStorage.getItem("confirmed orders")) === null ? [{
+        id: null,
+        restaurantName: null,
+        infoOfMainMeatItem: null,
```

Comment:
Try to find shorter ways to name your variables.
This can easily be named `meatItemInfo`